File: src/main/java/io/gaegul/ch01/puzzle/Character.java
package io.gaegul.ch01.puzzle;

import io.gaegul.ch01.puzzle.behavior.WeaponBehavior;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;

/**
 * 게임 캐릭터용 클래스
 */
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public abstract class Character {
	private WeaponBehavior weaponBehavior;

	public abstract void fight();
}

File: src/main/kotlin/io/gaegul/ch01/puzzle/Character.kt
package io.gaegul.ch01.puzzle

import io.gaegul.ch01.puzzle.behavior.WeaponBehavior

abstract class Character(
    private var weaponBehavior: WeaponBehavior
) {

    fun setWeapon(weaponBehavior: WeaponBehavior) {
        this.weaponBehavior = weaponBehavior
    }

    abstract fun fight()
}
